src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessHmcMessageService.java

```diff
@@ -60,6 +61,10 @@ public void processEventMessage(HmcMessage hmcMessage)
 
         SscsCaseData caseData = ccdCaseService.getCaseDetails(caseId).getData();
 
+        if (LISTED == hmcStatus) {
+            caseData.setDwpState(HEARING_DATE_ISSUED.getId());
+        }
```

Comment:
Do you mind extracting this into a method, probably within the class hearingUpdateService, as this method is already quite complex as it is.